### DISCUSSION TITLE
How can we set a DNS name for the Bootstrap when we start the listener in 9094 port with type loadbalancer for an internal load balancer type.

### DISCUSSION THREAD

**Discussion by sreejesh-radhakrishnan-db** [strimzi/strimzi-kafka-operator#6896] [Q&A]
How can we set a DNS name for the Bootstrap when we start the listener in 9094 port with type loadbalancer for an internal load balancer type. 

    type: loadbalancer
    tls: true
    authentication:
      type: tls
    configuration:
      bootstrap:
        annotations:
          service.beta.kubernetes.io/openstack-internal-load-balancer: "true"

**Comment by sreejesh-radhakrishnan-db** [comment]
when I try to access from outside the GKE 

SSL://my-cluster-kafka-tls-bootstrap.app.uat.gcp.db.com:9093

app.uat.gcp.db.com -this is my DNS domain 

Gettig the error: 
No resolvable bootstrap urls given in bootstrap.servers  -

**Reply by scholzj** [reply]
The name of the bootstrap loadbalancer is based on the information provided by Kubernetes. Whether you can influence it or not, depends on your infrastructure. Typically you cannot, but you can add additional DNS names for example using the External DNS names.

I'm also not sure how your questions relate -> you talk about GKE but use some OpenStack annotations? So I'm a bit confused. In general, it is up to your infrastructure to make the name resolvable and routable. Assuming you use OpenStack and create an internal loadbalancer, then maybe your application is not part of the same internal network?

**Comment by sreejesh-radhakrishnan-db** [comment]
@scholzj   sorry and well spotted.  

          annotations:
              cloud.google.com/load-balancer-type: "Internal"


if I want to use the External DNS names.  as you suggested .. any pointer or document as to how I can achieve that?

**Reply by scholzj** [reply]
ExternalDNS has really nothing to do with Strimzi. It just watches for annotations of objects such as services and assigns the DNS names to them in some external DNS system (e.g. in Amazon Route53, Google CLoudDNS etc.). ExternalDNS docs might be the best place to start.

**Reply by sreejesh-radhakrishnan-db** [reply]
Thank you,

 is it possible to ask KAFKA listener in 9094/9093 with ilb to use a specific IP for their Service to run on? Currently when I install the cluster what ever is available is getting used,


Idea is : want to  reserve a static IP  give DNS for that and use for the listener?

**Reply by scholzj** [reply]
Just try to go through the different options in the docs please. You can try: https://strimzi.io/docs/operators/latest/full/configuring.html#property-listener-config-lb-ip-reference ... but whether this is supported or not depends on your infrastructure.
